lib/ansible/modules/git.py

```diff
@@ -567,6 +567,36 @@ def get_submodule_versions(git_path, module, dest, version='HEAD'):
     return submodules
 
 
+def get_submodule_config(git_path, module, dest, submodule_name, config_name):
+    config_path = 'submodule.' + submodule_name + '.' + config_name
+    cmd = [git_path, 'config', '-f', '.gitmodules', '--get', config_path]
+    (rc, out, err) = module.run_command(cmd, cwd=dest)
+
+    out = out.rstrip('\n')
+    # if we can't resolve the actual branch, assume its master.
+    if not out:
+        out = "master"
+    return out
+
+
+def get_submodule_versions_from_remote(git_path, module, dest, submodule_revisions):
+    submodule_revisions_remote = {}
+    for submodule_name in submodule_revisions:
+        submodule_path = get_submodule_config(git_path, module, dest, submodule_name, 'path')
+        if not submodule_path:
+            module.fail_json(msg='Unable to detect path of submodule: %s' % submodule_name)
+
+        submodule_branch = get_submodule_config(git_path, module, dest, submodule_name, 'branch')
+        if not submodule_branch:
+            module.fail_json(msg='Unable to detect branch of submodule: %s' % submodule_name)
+
+        submodule_path = os.path.join(dest, submodule_path)
+        revision = get_version(module, git_path, submodule_path, '%s/%s' % ('origin', submodule_branch))
```

Comment:
Why is this hardcoding 'origin' instead of using the remote specified by the user?

Reply:
.gitmodules does not appear to list the origin.
```
[submodule "submodule3"]
        path = submodule3
        url = https://github.com/theunkn0wn1/ansible_test_submodules_subm3.git
        branch = user/theunkn0wn1/random_branch
[submodule "submodule4"]
        path = submodule4
        url = https://github.com/theunkn0wn1/ansible_test_submodules_subm4.git
        branch = main

```

Reply:
It wasn't hardcoded to `'origin'` before your change, the value from the module parameter `remote` was used.